lib/blumquist.rb

```diff
@@ -73,6 +71,8 @@ def define_getters
       # Nothing to do for primitive values
       elsif primitive_type?(type)
 
+      elsif all_primitive_types(type_def[:type])
```

Comment:
This is a bit odd, checking type_def again, after we have flattened it above, but it's the only way.